### PR TITLE
Remove unnecessary cast

### PR DESCRIPTION
**Pull request by fredj**
Found by the latest version of closure-compiler (v20150729)

### CODE REVIEW

**Comment by tschaub** [comment]
Looks good.  Please merge.

File: src/ol/feature.js [hunk 1]
```diff
@@ -302,13 +302,10 @@ ol.FeatureStyleFunction;
...
  */
 ol.Feature.createStyleFunction = function(obj) {
-  /**
-   * @type {ol.FeatureStyleFunction}
-   */
   var styleFunction;
 
   if (goog.isFunction(obj)) {
-    styleFunction = /** @type {ol.FeatureStyleFunction} */ (obj);
+    styleFunction = obj;
```
Comment (elemoine) [inline src/ol/feature.js]: The typing here and above makes sense to me. This is about making sure styleFunction is set to an ol.FeatureStyleFunction. Does the compiler complain about that when a specific flag is used?

**Comment by elemoine** [comment]
I just added a minor comment. Feel free to merge.

File: src/ol/feature.js (hunk 1 above)
Comment (fredj) [inline src/ol/feature.js]: With the `unnecessaryCasts` check enabled, the error is:
```
ERR! compile /home/fredj/public_html/ol3/src/ol/feature.js:311: ERROR - unnecessary cast
ERR! compile from: function (this:ol.Feature, number): (Array<(null|ol.style.Style)>|null)
ERR! compile to  : function (this:ol.Feature, number): (Array<(null|ol.style.Style)>|null)
ERR! compile     styleFunction = /** @type {ol.FeatureStyleFunction} */ (obj);
```

**Pull request merged**